src/header/header.scss

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../user-icon/index';
+@import '../icon/index';
+
+
+@mixin iui-header {
+  @include iui-reset;
+  @include iui-font-family;
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  height: $iui-baseline * 5;
+  transition: height $iui-speed-fast ease-out;
+  @include themed {
+    background-color: t(iui-color-background-1);
+    border-bottom: $iui-xxs solid t(iui-color-background-5);
+    color: t(iui-text-color);
+  }
+
+  > .iui-left,
+  > .iui-center,
+  > .iui-right {
+    display: flex;
+    align-items: center;
+    height: 100%;
+    width: 33%;
+  }
+
+  > .iui-left {
+    &:nth-last-child(2) {
+      width: unset;
+      overflow: hidden;
+    }
+    
+    > .iui-application,
+    > .iui-header-button {
+      overflow: hidden;
+      transition: height $iui-speed-fast ease-out;
+
+      > .iui-label {
+        overflow: hidden;
+        text-overflow: ellipsis;
+        transition: font-size $iui-speed-fast ease-out;
+
+        > * {
+          width: 100%;
+          overflow: hidden;
+          text-overflow: ellipsis;
+          text-align: left;
+        }
+      }
+
+      > .iui-brand-logo,
+      > .iui-thumbnail-area {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        height: 100%;
+
+        > .iui-icon,
+        > .iui-thumbnail {
+          width: $iui-icons-large;
+          height: $iui-icons-large;
+          transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out;
+        }
+      }
+    }
+
+    > .iui-application {
+      padding: 0;
+
+      > .iui-brand-logo {
+        width: $iui-l * 3 - 1;
+        flex-shrink: 0;
+      }
+
+      > .iui-label {
+        margin-right: 12px;
+        transition: margin-right $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out;
+      }
+    }
+
+    > .iui-header-button {
+      > .iui-thumbnail-area {
+        width: $iui-l;
+        margin-right: $iui-sm;
+
+        > .iui-thumbnail {
+          border-radius: $iui-border-radius;
+          background-position: center;
+          background-size: cover;
+          @include themed {
+            background-color: t(iui-color-background-3);
```

Comment:
Why it has `background-color`?

Reply:
The original idea was to have a background color & placeholder icon if no image is provided, but this is a good reminder that I didn't finish this part. 😅 I'll look into fixing this.
![Screen Shot 2021-04-27 at 5 08 17 PM](https://user-images.githubusercontent.com/849817/116312988-06580f00-a762-11eb-81a6-2a9bf94651ca.png)

Reply:
In my next push I copied the HTML structure & CSS from tile for thumbnails / placeholder icons.  Don't worry, there is no scaling transition applied here.
![Screen Shot 2021-05-04 at 12 04 10 PM](https://user-images.githubusercontent.com/849817/117034072-cd66ef80-acb7-11eb-9c1f-bdc6c4801498.png)